graylog2-server/src/main/java/org/graylog2/lookup/LookupModule.java

```diff
@@ -41,6 +42,7 @@ protected void configure() {
         binder().bind(UrlWhitelistNotificationService.class).in(Scopes.SINGLETON);
         binder().bind(AllowedAuxiliaryPathChecker.class).in(Scopes.SINGLETON);
 
+        bind(LookupTableConfigService.class).to(DBLookupTableConfigService.class).asEagerSingleton();
```

Comment:
Is there a particular reason you opted to use an eager singleton here instead of an optional binding (with default)?

Reply:
@danotorrey An optional binding would allow us to override a binding when running with a different application profile or loading a plugin. That would override the binding globally, though. In this case we always want to use the `DBLookupTableConfigService` to keep the lookup feature working as it is. We only want the option to create custom `LookupTableService` instances manually that don't depend on database backend configuration.

Reply:
Ah, of course. Makes sense.